python/pyspark/mllib/clustering.py

```diff
@@ -38,13 +38,120 @@
 from pyspark.mllib.util import Saveable, Loader, inherit_doc, JavaLoader, JavaSaveable
 from pyspark.streaming import DStream
 
-__all__ = ['KMeansModel', 'KMeans', 'GaussianMixtureModel', 'GaussianMixture',
-           'PowerIterationClusteringModel', 'PowerIterationClustering',
-           'StreamingKMeans', 'StreamingKMeansModel',
+__all__ = ['BisectingKMeansModel', 'BisectingKMeans', 'KMeansModel', 'KMeans',
+           'GaussianMixtureModel', 'GaussianMixture', 'PowerIterationClusteringModel',
+           'PowerIterationClustering', 'StreamingKMeans', 'StreamingKMeansModel',
            'LDA', 'LDAModel']
 
 
 @inherit_doc
+class BisectingKMeansModel(JavaModelWrapper):
+    """
+    .. note:: Experimental
+
+    A clustering model derived from the bisecting k-means method.
+
+    >>> data = array([0.0,0.0, 1.0,1.0, 9.0,8.0, 8.0,9.0]).reshape(4, 2)
+    >>> bskm = BisectingKMeans()
+    >>> model = bskm.train(sc.parallelize(data), k=4)
+    >>> p = array([0.0, 0.0])
+    >>> model.predict(p) == model.predict(p)
+    True
+    >>> model.predict(sc.parallelize([p])).first() == model.predict(p)
+    True
+    >>> model.k
+    4
+    >>> model.computeCost(array([0.0, 0.0]))
+    0.0
+    >>> model.k == len(model.clusterCenters)
+    True
+    >>> model = bskm.train(sc.parallelize(data), k=2)
+    >>> model.predict(array([0.0, 0.0])) == model.predict(array([1.0, 1.0]))
+    True
+    >>> model.k
+    2
+
+    .. versionadded:: 2.0.0
+    """
+
+    @property
+    @since('2.0.0')
+    def clusterCenters(self):
+        """Get the cluster centers, represented as a list of NumPy arrays."""
+        return [c.toArray() for c in self.call("clusterCenters")]
```

Comment:
It'd be nice to store the centers right after training the model.  I could imagine users calling this method within a closure.